File: leetcode/js/hash-divided-string.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var stringHash = function (s, k) {
  let result = "";

  for (let i = 0; i < s.length; i += k) {
    const sum = Array.from({ length: k }, (_, idx) => i + idx).reduce(
      (acc, cur) => s.charCodeAt(cur) - 97 + acc,
      0
    );

    result += String.fromCharCode((sum % 26) + 97);
  }

  return result;
};

File: leetcode/rust/hash-divided-string.rs
impl Solution {
    pub fn string_hash(s: String, k: i32) -> String {
        let k = k as usize;
        let mut result = String::new();

        for i in 0..(s.len() / k) {
            let sum = s[(i * k)..(i * k + k)]
                .bytes()
                .map(|b| b as u32 - 97)
                .sum::<u32>();
            let mod_num = (sum % 26) as u8;

            result.push((mod_num + 97) as char);
        }

        result
    }
}
